lib/grizzly/zipgateway/supervisor.ex

```diff
@@ -10,6 +10,18 @@ defmodule Grizzly.ZIPGateway.Supervisor do
 
   @zgw_eeprom_to_sqlite "/usr/bin/zgw_eeprom_to_sqlite"
 
+  @doc "Restarts the Z/IP Gateway process if it is running."
+  @spec restart_zipgateway ::
+          {:error, any()} | {:ok, :undefined | pid()} | {:ok, :undefined | pid(), any()}
+  def restart_zipgateway() do
+    if is_pid(GenServer.whereis(__MODULE__)) do
+      Supervisor.terminate_child(__MODULE__, MuonTrap.Daemon)
+      Supervisor.restart_child(__MODULE__, MuonTrap.Daemon)
```

Comment:
```suggestion
      _ = Supervisor.terminate_child(__MODULE__, MuonTrap.Daemon)
      {:ok, _} = Supervisor.restart_child(__MODULE__, MuonTrap.Daemon)
      :ok
```

I was debating whether this function should always raise on error - mostly since I couldn't think of a scenario where you'd call `restart_zipgateway/0` and having it not restart zipgateway would be ok.

If you agree, then my suggestion is to remove the `is_pid/1` check, use the code in the suggestion above and change the spec to always return `:ok`.